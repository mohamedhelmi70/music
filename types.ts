export type AppParamList = {
    Tab: undefined;
    Home: undefined;
    Track: {
        trackId: number;
    }
    Artists: undefined;
    Artist: {
        artistId: number;
    }
    Search: undefined;
    Albums: undefined;
    Album:  {
        albumId: number;
    }
};

export interface Album {
    album: string,
    id_album: number,
    cover: string,
    api_album: string,
    api_tracks: string
}

export interface Track {
    id_track: number;
    track: string;
    bpm: number;
    api_track: string;
    api_lyrics: string;
    artist: string;
}

export interface Artist {
    id_artist: number,
    artist: string,
    cover: string,
}