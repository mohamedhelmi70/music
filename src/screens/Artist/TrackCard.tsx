import React from 'react';
import {Pressable, StyleSheet} from 'react-native';
import {AppParamList, Track} from '../../../types';
import {StackNavigationProp} from '@react-navigation/stack';
import {colors, wp} from '../../constants';
import {Text, View} from '../../components/Themed';
import Icon from 'react-native-vector-icons/Ionicons';

interface TrackCardProps {
    track: Track;
    albumId: number;
    artistId: number;
    navigation?: StackNavigationProp<AppParamList, 'Artist'>;
}

function TrackCard({track, navigation, artistId, albumId}: TrackCardProps) {
    return (
        <Pressable
            style={{marginStart: wp('5%')}}
            onPress={() =>
                navigation?.navigate('Track', {trackId: track.id_track, title: track.track, albumId, artistId})
            }>
            <View style={styles.container}>
                <View lightColor={colors.light.gray} style={styles.image}>
                    <Icon name="md-musical-notes-outline" size={30} />
                </View>
                <View style={styles.cardRight}>
                    <Text style={styles.cardTitle} numberOfLines={1} weight="700" lightColor="#334443">
                        {track.track}
                    </Text>
                </View>
            </View>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    container: {
        width: wp('26.5%'),
        alignSelf: 'center',
        borderRadius: wp('3%'),
        marginBottom: wp('4%'),
    },
    image: {
        width: wp('26.5%'),
        height: wp('25.5%'),
        alignItems: 'center',
        justifyContent: 'center',
        borderTopLeftRadius: wp('3%'),
        borderTopRightRadius: wp('3%'),
    },
    cardRight: {
        marginLeft: wp('2%'),
        paddingVertical: wp('3%'),
    },
    cardTitle: {
        fontSize: wp('2.5%'),
        maxWidth: wp('40%'),
        width: wp('25.8%'),
    },
});

export default TrackCard;
