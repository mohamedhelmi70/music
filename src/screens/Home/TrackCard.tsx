import React from 'react';
import {Pressable, StyleSheet} from 'react-native';
import {AppParamList, Track} from '../../../types';
import {StackNavigationProp} from '@react-navigation/stack';
import {colors, wp} from '../../constants';
import {Text, View} from '../../components/Themed';
import Icon from 'react-native-vector-icons/Ionicons';

interface TrackCardProps {
    track: Track;
    albumId: number;
    artistId: number;
    navigation: StackNavigationProp<AppParamList, 'Home'>;
}

function TrackCard({track, navigation, albumId, artistId}: TrackCardProps) {
    return (
        <Pressable
            onPress={() =>
                navigation.navigate('Track', {trackId: track.id_track, title: track.track, albumId, artistId})
            }>
            <View lightColor="#FFF" style={styles.container}>
                <View lightColor={colors.light.gray} style={styles.image}>
                    <Icon name="md-musical-notes-outline" size={30} />
                </View>
                <View style={styles.cardRight} lightColor="#FFF">
                    <Text style={styles.cardTitle} weight="700" lightColor="#334443">
                        {track.track}
                    </Text>

                    <Text style={styles.caption} font="Roboto-Medium" weight="500" lightColor={colors.light.dark}>
                        {track.artist}
                    </Text>
                </View>
            </View>
        </Pressable>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        width: wp('90%'),
        alignSelf: 'center',
        borderRadius: wp('3%'),
        marginBottom: wp('4%'),

        flexDirection: 'row',
    },
    image: {
        width: wp('25%'),
        height: wp('25%'),
        alignItems: 'center',
        justifyContent: 'center',
        borderTopLeftRadius: wp('3%'),
        borderBottomLeftRadius: wp('3%'),
    },
    cardRight: {
        marginLeft: wp('4%'),
        paddingVertical: wp('3%'),
    },
    cardTitle: {
        fontSize: wp('3.6%'),
        maxWidth: wp('40%'),
        width: wp('40%'),
    },
    caption: {
        fontSize: wp('3%'),
        maxWidth: wp('40%'),
        width: wp('40%'),
        marginTop: wp('1.5%'),
    },
});

export default TrackCard;
