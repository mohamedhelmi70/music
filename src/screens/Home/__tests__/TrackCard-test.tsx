import React from 'react';
import renderer from 'react-test-renderer';
import {Track} from '../../../../types';
import TrackCard from '../TrackCard';

// Describing a test suite
describe('<TrackCard />', () => {
    const track: Track = {
        id_track: 11,
        track: 'track name',
        artist: 'artist name',
        album: 'album name',
        api_track: 'hhtps://music.com/track',
        api_lyrics: 'hhtps://music.com/lyrics',
        haslyrics: false,
        bpm: 12,
    };
    test('Renders TrackCard correctly', () => {
        const tree = renderer.create(<TrackCard track={track} albumId={1212} artistId={2323} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
