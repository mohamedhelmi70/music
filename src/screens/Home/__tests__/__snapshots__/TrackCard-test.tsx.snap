// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackCard /> Renders TrackCard correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        Object {
          "alignSelf": "center",
          "borderRadius": 22.5,
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 30,
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#C1C6C7",
          },
          Object {
            "alignItems": "center",
            "borderBottomLeftRadius": 22.5,
            "borderTopLeftRadius": 22.5,
            "height": 187.5,
            "justifyContent": "center",
            "width": 187.5,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Ôêñ
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFF",
          },
          Object {
            "marginLeft": 30,
            "paddingVertical": 22.5,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#334443",
              "fontFamily": "Roboto",
              "fontWeight": "700",
              "textAlign": "left",
            },
            Object {
              "fontSize": 27,
              "maxWidth": 300,
              "width": 300,
            },
          ]
        }
      >
        track name
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#8B899F",
              "fontFamily": "Roboto",
              "fontWeight": "500",
              "textAlign": "left",
            },
            Object {
              "fontSize": 22.5,
              "marginTop": 11.5,
              "maxWidth": 300,
              "width": 300,
            },
          ]
        }
      >
        artist name
      </Text>
    </View>
  </View>
</View>
`;
